### PR TITLE
Delete one session by Id in order to revoke access 

### PR DESCRIPTION
**Pull request by karangwaajika**
# 🚀 Pull Request: Revoke/Delete User Session (Admin)

### 🎫 Jira Ticket  
[🔗 TRA-110: Allow only admin to view active user sessions](https://amali-tech.atlassian.net/browse/TRA-15)  

---

## ✅ Summary

This PR introduces functionality that allows **admins to revoke (delete) active user sessions**.

---

## 📌 Features Added

- [x] Admin can revoke (delete) any user session
- [x] New REST endpoint for session deletion
- [x] Swagger documentation for the new endpoint
- [x] Logging for session deletion operations
- [x] Unit and service-level tests added

---

## 🔍 Technical Details

- Leverages `UserSessionRepository.deleteById(sessionId)`
- Service layer checks for session existence before deletion
- Returns:
  - `204 No Content` on success
  - `404 Not Found` if session ID does not exist

---

## 🧪 Tests

- [x] `SessionServiceTests`: verifies deletion logic
- [x] Controller tests for endpoint behavior (if applicable)
- [x] Edge case: attempt to delete non-existent session

---

## 📖 Swagger

- [x] `DELETE /sessions/{sessionId}` documented
- [x] Includes request/response schema and HTTP status codes

---

## 🪵 Logging

- [x] `INFO`: When a session is successfully revoked
- [x] `WARN`: If session ID not found during deletion attempt

---

## 🚧 Next Task

> Implement **session filtering and pagination**, including:
> - IP address
> - Device type
> - Created date range
> - User ID

---

## 📎 References

- Jira Ticket: `GTRS-1234`
- Linked Issue: `#57`

### CODE REVIEW

File: src/main/java/com/talentradar/user_service/controller/SessionController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@RequiredArgsConstructor
+@Tag(name = "Session Controller", description = "Manage all user's session api")
+public class SessionController {
+    private static final Logger logger = LoggerFactory.getLogger(SessionController.class);
+    private final SessionService sessionService;
+
+    @GetMapping(name = "fetchActiveSessions", path = "/sessions")
+    @Operation(summary = "Fetch all active session",
+            description = "This end point allow only admin to view all of the active session")
+//    @PreAuthorize("hasRole('Admin')")
```
Comment (nccharles) [inline src/main/java/com/talentradar/user_service/controller/SessionController.java]: Avoid pushing commented code. Can explain why removed it?

**Review by nccharles — CHANGES_REQUESTED** [review]
Kindly resolve the message a added and pull new changes from the new merged one

File: src/main/java/com/talentradar/user_service/controller/SessionController.java (hunk 1 above)
Comment (karangwaajika) [reply src/main/java/com/talentradar/user_service/controller/SessionController.java]: This is not commented; it's the Swagger UI documentation.

Comment (nccharles) [reply src/main/java/com/talentradar/user_service/controller/SessionController.java]: //    @PreAuthorize("hasRole('Admin')")

Comment (karangwaajika) [reply src/main/java/com/talentradar/user_service/controller/SessionController.java]: Alright

Comment (karangwaajika) [reply src/main/java/com/talentradar/user_service/controller/SessionController.java]: This is when testing some cases. I just forgot to rollback.

**Review by nccharles — APPROVED** [review]
